src/plugins/data/public/search/fetch/extract_warnings.ts

```diff
@@ -0,0 +1,89 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0 and the Server Side Public License, v 1; you may not use this file except
+ * in compliance with, at your election, the Elastic License 2.0 or the Server
+ * Side Public License, v 1.
+ */
+
+import { estypes } from '@elastic/elasticsearch';
+import { i18n } from '@kbn/i18n';
+
+/**
+ * The format of warning notifications sourced from info in a search response body
+ */
+export interface SearchResponseWarningNotification {
+  /**
+   * Title for the heading of the warning toast notification
+   */
+  title?: string;
+  /**
+   * Text for the body of the warning toast notification
+   */
+  text?: string;
+}
+
+/**
+ * The format of warning data which may be preset in search response body (shard failures, timeouts).
+ * @public
+ */
+export interface SearchResponseWarnings {
+  notifications?: {
+    timedOut?: SearchResponseWarningNotification;
+    shardFailures?: SearchResponseWarningNotification;
+  };
+  types?: string[]; // shard stats can have multiple shard failure reasons, such as "illegal_argument_exception", etc
```

Comment:
seems just array of types is not going to be very usefull ?

Reply:
Hi @ppisljar what do you suggest?

Reply:
- The latest code adds a different warning for each shard failure "reason"
- The `type` field is `'timed_out' | 'generic_shard_warning' | estypes.ShardFailure['reason']['reason'];`

https://github.com/elastic/kibana/commit/3a22146c3c2889dc5eb8572f8725e5a91bdc02d6